src/parameters/define_parameter_test.ts

```diff
@@ -0,0 +1,40 @@
+import { DefineParameter } from "./define_parameter.ts";
+import SchemaTypes from "../schema/schema_types.ts";
+import { assert, IsExact } from "../dev_deps.ts";
+
+Deno.test("DefineParameter should allow for object property names to be specified in the required field", () => {
+  const obj = DefineParameter({
+    type: SchemaTypes.object,
+    properties: {
+      aString: {
+        type: SchemaTypes.string,
+      },
+      anOptionalString: {
+        type: SchemaTypes.string,
+      },
+    },
+    required: ["aString"],
+    additionalProperties: true,
+  });
+  assert<IsExact<typeof obj.required, "aString"[]>>(true);
+});
+
+/*
+TODO: DefineParameter fails to constrain the required field to property names :(
```

Comment:
Really wish we could solve this issue 😢

Reply:
Agreed, the solution we found leads to another inexplicable issue where devs may need to instantiate a variable that uses `DefineParameter` outside of a `DefineFunction` call, which seems unacceptable to me